flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -1258,8 +1258,8 @@
                 CastTestSpecBuilder.testCastTo(ARRAY(BIGINT().nullable()))
                         .fromCase(
                                 ARRAY(INT().nullable()),
-                                new GenericArrayData(new Object[] {1, null, 2}),
-                                new GenericArrayData(new Object[] {1L, null, 2L})),
+                                new GenericArrayData(new Integer[] {1, null, 2}),
+                                new GenericArrayData(new Long[] {1L, null, 2L})),
```

Comment:
it seems that the case would not failed with Object array and the change above in my local project, there may be some bug  on the validation of the test spec.

Reply:
But this is the actual issue, right? `GenericArrayData` requires boxed integer arrays to correctly use `toIntArray`, right?

Reply:
yes.

Reply:
Ok then I think this is good to go